sam.c

```diff
@@ -522,6 +522,7 @@ int sam_index_build2(const char *fn, const char *fnidx, int min_shift)
         idx = bam_index(fp->fp.bgzf, min_shift);
         if (idx) {
             ret = hts_idx_save_as(idx, fn, fnidx, (min_shift > 0)? HTS_FMT_CSI : HTS_FMT_BAI);
+            ret = ret ? -4 : 0;
```

Comment:
Please do 😄

Me, I'm pro-`?:` in general, but in this case `ret = ret ? -4 : 0` bashes positive return values over the head, and to validate the change readers have to think about whether positive return values should be bashed to -4 (surely not!) or 0, and then go off and check exactly what `hts_idx_save_as` can return this week, and only then realise that positive return values (currently) won't occur.

Which is all a lot more work than just applying the general htslib rule that `retcode<0` ⇒ error.